src/Units/KaijuActions.java

```diff
@@ -214,25 +250,45 @@ public static void enqueueSingleTileEvents(MapMaster gameMap, int startingMP, Un
         {
           Environment oldEnvirons = location.getEnvironment();
           Environment newEnvirons = Environment.getTile(oldEnvirons.terrainType.getBaseTerrain(), oldEnvirons.weatherType);
-          crushEvents.add(new MapChangeEvent(location.getCoordinates(), newEnvirons));
+          crush.events.add(new MapChangeEvent(location.getCoordinates(), newEnvirons));
+
+          // Overwrite existing popup, but don't spawn one
+          if( null != tileToast )
+            tileToast = "WRECK";
+
           if( stomperType.regenOnBuildingKill )
-            crushEvents.add(new HealUnitEvent(kaijuState.unit, 2, null, true));
+          {
+            int heal = 2;
+            crush.events.add(new HealUnitEvent(kaijuState.unit, heal, null, true));
+            kaijuState.alterHP(heal); // previewed by caller
+          }
           if( stomperType.chargeOnBuildingKill && location.isCaptureable() )
-            crushEvents.add(new CommanderEnergyChangeEvent(kaijuState.CO, 1));
-
-          if( Utils.willLoseFromLossOf(gameMap, location) )
           {
-            crushEvents.add(new ArmyDefeatEvent(location.getOwner().army));
+            crush.events.add(new CommanderEnergyChangeEvent(kaijuState.CO, 1));
+            tileToast = "ENERGY";
           }
+
+          // Icky, but I don't really wanna change this function signature
+          if( gameMap instanceof MapMaster )
+            if( Utils.willLoseFromLossOf((MapMaster) gameMap, location) )
+            {
+              crush.events.add(new ArmyDefeatEvent(location.getOwner().army));
+            }
```

Comment:
This is icky. And you are changing the function signature; it accepted a `MapMaster` before.

Reply:
I didn't change the function signature of `Utils.willLoseFromLossOf()`, and I don't really want to because it's used to decide when to end the game - you want complete information for that.